packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -2129,6 +2129,59 @@ void main() {
     // No exception should be thrown.
     expect(tester.takeException(), isNull);
   });
+
+  // This is a regression test for https://github.com/flutter/flutter/issues/147076.
+  testWidgets('Text field does not overflow parent', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: SizedBox(
+          width: 300,
+          child: DropdownMenu<int>(
+            dropdownMenuEntries: <DropdownMenuEntry<int>>[
+              DropdownMenuEntry<int>(
+                value: 0,
+                label: 'This is a long text that is multiplied by 4 so it can overflow. ' * 4,
+              ),
+            ],
+          ),
+        ),
+      ),
+    ));
+
+    await tester.pump();
+    final RenderBox box = tester.firstRenderObject(find.byType(TextField));
+    expect(box.size.width, 300.0);
+  });
+
+  // This is a regression test for https://github.com/flutter/flutter/issues/147173.
+  testWidgets('Text field with large helper text can be selected', (WidgetTester tester) async {
+    await tester.pumpWidget(
+     const MaterialApp(
+      home: Scaffold(
+        body: Center(
+          child: DropdownMenu<int>(
+            hintText: 'Hint text',
+            helperText: 'Menu Helper text',
+            inputDecorationTheme: InputDecorationTheme(
+              helperMaxLines: 2,
+              helperStyle: TextStyle(fontSize: 30),
+            ),
+            dropdownMenuEntries: <DropdownMenuEntry<int>>[
+              DropdownMenuEntry<int>(
+                value: 0,
+                label: 'MenuEntry 1',
+              ),
+            ],
+          ),
+        ),
+      ),
+    ));
+
+  await tester.pump();
+  await tester.tapAt(tester.getCenter(find.text('Hint text')));
+  await tester.pumpAndSettle();
+  expect(find.byType(MenuItemButton), findsNWidgets(2));
```

Comment:
```suggestion
    // One is layout for the _DropdownMenuBody, the other one is the real button item in the menu.
    expect(find.widgetWithText(MenuItemButton, label), findsNWidgets(2));
```
Consider reusing a comment seen in another test from this file which clarifies why we are looking for two occurrences. Also consider relying on `find.widgetWithText` for a more precise finder (doing so the menu item label is repeated so introducing a const for the label would be great).